src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -296,15 +322,27 @@ bool LibraryCallKit::inline_vector_nary_operation(int n) {
     }
   }
 
-  // TODO When mask usage is supported, VecMaskNotUsed needs to be VecMaskUseLoad.
-  if ((sopc != 0) &&
-      !arch_supports_vector(sopc, num_elem, elem_bt, is_vector_mask(vbox_klass) ? VecMaskUseAll : VecMaskNotUsed)) {
-    if (C->print_intrinsics()) {
-      tty->print_cr("  ** not supported: arity=%d opc=%d vlen=%d etype=%s ismask=%d",
-                    n, sopc, num_elem, type2name(elem_bt),
-                    is_vector_mask(vbox_klass) ? 1 : 0);
+  if (VectorNode::is_rotate_opcode(opc)) {
+    if ((sopc != 0) &&
+        !arch_supports_vector_rotate(sopc, num_elem, elem_bt, false)) {
+      if (C->print_intrinsics()) {
+        tty->print_cr("  ** not supported: arity=%d opc=%d vlen=%d etype=%s ismask=%d",
+                      n, sopc, num_elem, type2name(elem_bt),
+                      is_vector_mask(vbox_klass) ? 1 : 0);
+      }
+      return false; // not supported
+    }
+  } else {
+    // TODO When mask usage is supported, VecMaskNotUsed needs to be VecMaskUseLoad.
+    if ((sopc != 0) &&
+        !arch_supports_vector(sopc, num_elem, elem_bt, is_vector_mask(vbox_klass) ? VecMaskUseAll : VecMaskNotUsed)) {
```

Comment:
Could we not call arch_supports_vector_rotate from arch_supports_vector?

Reply:
DONE